### ISSUE TITLE
add a possibility for displaying the "Committed Files"-Tree on top of the diff-view

### ISSUE THREAD

**Issue by arminfuerst** [Murmele/Gittyup#105]
In my opinion, the pane to see the diff is too small to have a comfortable view, here is a screenshot how it currently looks on my screen:
![image](https://user-images.githubusercontent.com/29121472/155227720-46e0c564-47aa-42e6-a9d7-19f844080369.png)
Resizing of the pane doesn't really help as it would make the list of commits uncomfortable small.

I think a view having the tree on top of the diff view would be more comfortable. I think about a design looking like something like this:
![image](https://user-images.githubusercontent.com/29121472/155230161-72083af2-0a84-4219-9576-68f1135fe5ed.png)

The view with staged / unstaged files could be left-to-right (diff-view and commit message in the next image is just copied from above, the content doesn't make sense in this context, the graphics is only to give an impression regarding the staged/unstaged files):
![image](https://user-images.githubusercontent.com/29121472/155230097-817a8c0e-fe9a-4b2b-90b6-449b3aeefa73.png)

**Comment by arminfuerst** [comment]
> @arminfuerst are you using your monitor vertical?

No. I assume you are asking because of the resolution, so the explanation might be that I prefer multiple displays and currently have a resolution of 1280x1024 on all my displays.

**Comment by Murmele** [comment]
> No. I assume you are asking because of the resolution, so the explanation might be that I prefer multiple displays and currently have a resolution of 1280x1024 on all my displays.

So you have multiple windows on one display, otherwise it is not possible that the width is smaller than the height

**Comment by arminfuerst** [comment]
> > No. I assume you are asking because of the resolution, so the explanation might be that I prefer multiple displays and currently have a resolution of 1280x1024 on all my displays.
> 
> So you have multiple windows on one display, otherwise it is not possible that the width is smaller than the height

No, gittyup is running full screen, but the (first) screenshot of this issue is just a part of the window, that's why it's width is smaller than the height. The other "screenshots" are created using GIMP, just in case my description would not be clear enough...

**Comment by Murmele** [comment]
> No, gittyup is running full screen, but the (first) screenshot of this issue is just a part of the window, that's why it's width is smaller than the height. The other "screenshots" are created using GIMP, just in case my description would not be clear enough...

No it's clear. It shouldn't be that a problem to implement

**Comment by arminfuerst** [comment]
> > No, gittyup is running full screen, but the (first) screenshot of this issue is just a part of the window, that's why it's width is smaller than the height. The other "screenshots" are created using GIMP, just in case my description would not be clear enough...
> 
> No it's clear. It shouldn't be that a problem to implement

Thanks - that's great to read :)

**Comment by mwerle** [comment]
Just to add a thought:  instead of hard-coding various different panel configurations, I think it would be better to re-design the UI so that the  user can move/resize/snap the various items (diff, committed files, commit details, log, commit list, etc) as they like. Ideally even being able to snap multiple items on top of each other instead of side-by-side or top-to-bottom. Items snapped on top of each other would create a tabbed view.

**Comment by arminfuerst** [comment]
> Just to add a thought: instead of hard-coding various different panel configurations, I think it would be better to re-design the UI so that the user can move/resize/snap the various items (diff, committed files, commit details, log, commit list, etc) as they like. Ideally even being able to snap multiple items on top of each other instead of side-by-side or top-to-bottom. Items snapped on top of each other would create a tabbed view.

Of course this would be an even better solution. My only concern is, that this would be even more effort and would delay this issue even longer. I'm desperately waiting for this, as Gittyup is the best available git client in my opinion, but when using a 4:3 display, the current view is a big limitation.
